### PR TITLE
INS-84: Class actions

### PR DESCRIPTION
**Pull request by andreyromancev**
Closes INS-84

**- What I did**
Implemented class actions in Artifact Manager

**- How to verify it**
Run tests

**- Description for the changelog**
Implemented basic interface for processing module interacting with storage

### CODE REVIEW

File: ledger/artifactmanager/artifactmanager_test.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	assert.NotNil(t, err)
+}
+
+func TestDeactivateClassVerifiesClassIsActive(t *testing.T) {
+	ledger := LedgerMock{map[record.ID]record.Record{}, map[record.ID]*index.Lifeline{}}
+	manager := LedgerArtifactManager{storer: &ledger}
+	requestRef := record.Reference{Domain: record.ID{1}, Record: record.ID{2}}
+	classRef, _ := ledger.AddRecord(&record.ClassActivateRecord{})
+	deactivateRef, _ := ledger.AddRecord(&record.DeactivationRecord{})
+	ledger.SetIndex(classRef.Record, &index.Lifeline{
```
Comment (golangci) [inline ledger/artifactmanager/artifactmanager_test.go]: Error return value of `ledger.SetIndex` is not checked

File: ledger/artifactmanager/artifactmanager_test.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+	_, err := manager.DeactivateClass(requestRef, classRef)
+	assert.NotNil(t, err)
+}
+
+func TestDeactivateClassCreatesDeactivateRecord(t *testing.T) {
+	ledger := LedgerMock{map[record.ID]record.Record{}, map[record.ID]*index.Lifeline{}}
+	manager := LedgerArtifactManager{storer: &ledger}
+	requestRef := record.Reference{Domain: record.ID{1}, Record: record.ID{2}}
+	classRef, _ := ledger.AddRecord(&record.ClassActivateRecord{})
+	ledger.SetIndex(classRef.Record, &index.Lifeline{
```
Comment (golangci) [inline ledger/artifactmanager/artifactmanager_test.go]: Error return value of `ledger.SetIndex` is not checked

File: ledger/artifactmanager/artifactmanager_test.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+	manager := LedgerArtifactManager{storer: &ledger}
+	requestRef := record.Reference{Domain: record.ID{1}, Record: record.ID{2}}
+	migrations := []record.MemoryMigrationCode{
+		record.MemoryMigrationCode{
+			MigrationCodeRecord: record.Reference{Record: record.ID{1, 2, 3}},
+		},
+	}
+	classRef, _ := ledger.AddRecord(&record.ClassActivateRecord{})
+	deactivateRef, _ := ledger.AddRecord(&record.DeactivationRecord{})
+	ledger.SetIndex(classRef.Record, &index.Lifeline{
```
Comment (golangci) [inline ledger/artifactmanager/artifactmanager_test.go]: Error return value of `ledger.SetIndex` is not checked

**Comment by ita-sammann** [comment]
ArtifactManager interface does not match real implementation. It's OK for now.

**Pull request merged**